Simplify Notification::RemovingComponent invocation

BeniCheni opened the pull request:
Fixes #1993.

After analyzing #1993 that led to #1283 & the source of #1639, the hypothesis is to simplify to just `Notification::RemovingComponent` invocation in "src/dist/manifestation.rs" and "src/dist/notification.rs", based on the understanding of the goal is to ultimately just display:

> info: removing previous version of component ***

By such simplification, also able to delete unneeded `RemovingOldComponent` code. Looking forward to review feedback & mentioning to update the PR, if my analysis or hypothesis is correctly.

P.S: I also tried to understand this comment from the 1st comment in #1993:

> Strangely enough, the relevant tests pass.

https://github.com/rust-lang/rustup.rs/blob/489fa585117dc0ac283f247f3c9a829fd32442af/src/dist/manifestation.rs#L133

I'm speculating that the `altered` variable checks for default dist server, and guessing that when tests run, the local env would somehow be considered truthy for the `altered` flag to call original `RemovingOldComponent` to land to the output code for "info: removing old component", and the tests would pass/

While running the "_prod dist_" build, the `altered` flag was considered falsy, and it would call the original `RemovingComponent` to land to the output code for "info: removing component".

tesuji committed:
Replace markdown with pulldown-cmark

BeniCheni committed:
Update notification for removing component

Correct notification for removing component

Revert edit in src/dist/manifestation.rs

thedrow reviewed (changes requested):
Thank you for working on this.

The reason for adding `RemovingOldComponent` was because you can also `rustup component remove`.
In this case, you are not removing an old component to replace it with a new one. You are simply removing it since you no longer need it.
I don't think that this behaviour is correct from a UX perspective.

BeniCheni commented:
Thanks, @thedrow. Will take a look and correct.

kinnison committed:
command: Pass Ctrl+C on in Windows

Attempt to replicate the functionality from
<https://github.com/rust-lang/cargo/pull/6004>
in order to pass Ctrl+C on in Windows

Signed-off-by: Daniel Silverstone <dsilvers@digital-scurf.org>

kinnison committed:
Disable Windows builds through Travis

Signed-off-by: Daniel Silverstone <dsilvers@digital-scurf.org>

BeniCheni commented:
Will open a new PR.

BeniCheni commented:
Hi, @thedrow , created a new PR (#2016) w. correction. PTAL over in the new PR? Thanks!
